cognite/extractorutils/uploader/files.py

```diff
@@ -408,12 +409,22 @@ def _get_file_upload_request(
         self, url_str: str, stream: BinaryIO, size: int, mime_type: Optional[str] = None
     ) -> Request:
         url = URL(url_str)
+        base_url = URL(self.cdf_client.config.base_url)
+
+        if url.host == base_url.host:
+            upload_url = url
+        else:
+            parsed_url = urlparse(url_str)
+            parsed_base_url = urlparse(self.cdf_client.config.base_url)
+            replaced_upload_url = parsed_url._replace(netloc=parsed_base_url.netloc).geturl()
```

Comment:
Using an internal method is not ideal, is there no public alternative?

Reply:
Nope. Tried to find it, but even in the urllib.parse docs they use the internal one as example =P 
https://docs.python.org/3/library/urllib.parse.html